### ISSUE TITLE
Page times out when the page content matches a glossary term and no forbiddenParentTags are set

### ISSUE THREAD

**Issue by RauchF** [featdd/dpn_glossary#34]
I'm using TYPO3 CMS 7.2.0 and dpn_glossary 2.2.4, with a (apart from Flux and Fluidcontent being installed) otherwise vanilla site.

When I navigate to a page that contains a word that matches a glossary term, the page does not load, but times out after the maximum execution time is reached, with the following stacktrace:

```
Fatal error: Maximum execution time of 600 seconds exceeded in /var/www/vhosts/intranet/data/typo3conf/ext/dpn_glossary/Classes/Service/WrapperService.php on line 145

Call Stack:
    0.0005     233632   1. {main}() /home/typo3/typo3_src/typo3_src-7.2.0/index.php:0
    0.0254    1452016   2. TYPO3\CMS\Core\Core\Bootstrap-&gt;run() /home/typo3/typo3_src/typo3_src-7.2.0/index.php:33
    0.0818    2956776   3. TYPO3\CMS\Frontend\RequestHandler-&gt;handleRequest() /home/typo3/typo3_src/typo3_src-7.2.0/typo3/sysext/core/Classes/Core/Bootstrap.php:191
    0.8549   17282480   4. TYPO3\CMS\Frontend\Controller\TypoScriptFrontendController-&gt;generatePage_postProcessing() /home/typo3/typo3_src/typo3_src-7.2.0/typo3/sysext/frontend/Classes/RequestHandler.php:214
    0.8549   17282832   5. TYPO3\CMS\Core\Utility\GeneralUtility::callUserFunction() /home/typo3/typo3_src/typo3_src-7.2.0/typo3/sysext/frontend/Classes/Controller/TypoScriptFrontendController.php:3337
    0.8557   17289992   6. call_user_func_array:{/home/typo3/typo3_src/typo3_src-7.2.0/typo3/sysext/core/Classes/Utility/GeneralUtility.php:4155}() /home/typo3/typo3_src/typo3_src-7.2.0/typo3/sysext/core/Classes/Utility/GeneralUtility.php:4155
    0.8557   17291424   7. DPN\DpnGlossary\Service\WrapperService-&gt;contentParser() /home/typo3/typo3_src/typo3_src-7.2.0/typo3/sysext/core/Classes/Utility/GeneralUtility.php:4155
  601.1411   21096992   8. DOMNode-&gt;getNodePath() /var/www/vhosts/intranet/data/typo3conf/ext/dpn_glossary/Classes/Service/WrapperService.php:145
```

This also happens when the term is not inside a Fluidcontent element but a regular CSC element.

Thanks in advance.

**Comment by featdd** [comment]
Would you mind to tell me your PHP version and check your HTML for correctness?
I can't reproduce this error but it could be some kind of endless loop when DOMNode tries to resolve its path in the xml tree and a closing tag is missing or something.

Greetings
featdd

**Comment by RauchF** [comment]
I'm using PHP 5.5.23.

I've checked the HTML, it contains no errors according to https://validator.w3.org/nu/ .

**Comment by featdd** [comment]
Hm thought it could have been some broken html otherwise I can't explain the crash at getNodePath for now. It just resolves the xml tree so I can check if the tag I want to parse has parents which are not allowed. Even on big html files it should'nt exceed the maximum execution time :-/

**Comment by RauchF** [comment]
Here's my constants for dpn_glossary, if it helps. You'll notice I set the forbiddenParentTags to an empty string:
```
plugin.tx_dpnglossary.settings.parsingTags = p,span,a,h1,h2,h3,h4,h5,h6
plugin.tx_dpnglossary.settings.forbiddenParentTags =
plugin.tx_dpnglossary.persistence.storagePid = 24
```

**Comment by featdd** [comment]
Ah thats the snag. you don't want to parse a tags and want them to be forbidden as parent, because you dont want a link generated in a link.

I will take a look into my code to catch if no forbiddenparenttags were set but I highly recommened to keep "a" as default and remove the "a" from your parsingtags.

**Comment by RauchF** [comment]
Ah, that's the thing. I'm not planning to use a detail or list view (and thus have no need for links), but basically just <abbr>, <acronym>, etc with their title attributes (as is possible with a21glossary from the government package). Is that behaviour supported? We use a21glossary on a couple sites at the moment, but we're looking to move to a more actively developed and supported extension (a21glossary only works with the compatibility6 extension currently).

**Comment by featdd** [comment]
You can configure the behaviour as you like over the extension settings. Just look at the standard configuration and you'll see.

This extension is actively developed, so if you have any questions just ask.

**Comment by featdd** [comment]
Solved in f7d7ef2c1cfa0e9e88fb97269fe2a29565917957
